src/common/servers/serverManager.ts

```diff
@@ -0,0 +1,426 @@
+            Config.localTeams.sort((a, b) => a.order - b.order).forEach((team) => {
+                const id = this.initServer(team, false);
+                serverOrder.push(id);
+            });
+        }
+        this.filterOutDuplicateTeams();
+        this.serverOrder = serverOrder;
+        if (Config.lastActiveTeam) {
+            this.currentServerId = this.serverOrder[Config.lastActiveTeam];
+        } else {
+            this.currentServerId = this.serverOrder[0];
+        }
+    }
+
+    private filterOutDuplicateTeams = () => {
+        const servers = [...this.servers.keys()].map((key) => ({key, value: this.servers.get(key)!}));
+        const uniqueServers = new Set();
+        servers.forEach((server) => {
+            if (uniqueServers.has(`${server.value.name}:${server.value.url}`)) {
+                this.servers.delete(server.key);
+            } else {
+                uniqueServers.add(`${server.value.name}:${server.value.url}`);
+            }
+        });
+    }
+
+    private initServer = (team: ConfigTeam, isPredefined: boolean) => {
+        const server = new MattermostServer(team, isPredefined);
+        this.servers.set(server.id, server);
+
+        log.debug('initialized server', server.id);
+
+        const tabOrder: string[] = [];
+        team.tabs.sort((a, b) => a.order - b.order).forEach((tab) => {
+            const tabView = this.getTabView(server, tab.name, tab.isOpen);
+            log.debug('initialized tab', tabView.id);
+
+            this.tabs.set(tabView.id, tabView);
+            tabOrder.push(tabView.id);
+        });
+        this.tabOrder.set(server.id, tabOrder);
+        if (typeof team.lastActiveTab !== 'undefined') {
+            this.lastActiveTab.set(server.id, tabOrder[team.lastActiveTab]);
+        }
+        return server.id;
+    }
+
+    private getFirstOpenTabForServer = (serverId: string) => {
+        const tabOrder = this.tabOrder.get(serverId);
+        if (!tabOrder) {
+            throw new Error(`Cannot find tabs for server id ${serverId}`);
+        }
+        const openTabs = this.getOrderedTabsForServer(serverId).filter((tab) => tab.isOpen);
+        const firstTab = openTabs[0];
+        if (!firstTab) {
+            throw new Error(`No tabs open for server id ${serverId}`);
+        }
+        return firstTab;
+    }
+
+    private persistServers = async (lastActiveTeam?: number) => {
+        this.emit(SERVERS_UPDATE);
+
+        const localServers = [...this.servers.values()].
+            filter((server) => !server.isPredefined).
+            map((server) => this.toConfigTeam(server));
+        await Config.setServers(localServers, lastActiveTeam);
+    }
+
+    private getLastActiveTab = (serverId: string) => {
+        let lastActiveTab: number | undefined;
+        if (this.lastActiveTab.has(serverId)) {
+            const index = this.tabOrder.get(serverId)?.indexOf(this.lastActiveTab.get(serverId)!);
+            if (typeof index !== 'undefined' && index >= 0) {
```

Comment:
a shorthand can be

```suggestion
            if (index != null && index >= 0) {
```

Reply:
I don't think this works here, because `null` and `undefined` are different identifiers. `Map.get()` returns undefined if the element doesn't exist, so index would be `undefined`.

I think this might get around TypeScript, but isn't technically correct. 0/5 on whether we change or not.

Reply:
`!= null` works in its own way (JavaScript magic) and checks either for `undefined` or `null`.

Be aware of `!=` instead of `!==`.

Reply:
Ugh, Javascript lol.
I think I'm going to leave it as is to avoid confusion, this is more explicit.